src/main/java/sx/blah/discord/handle/audit/AuditLog.java

```diff
@@ -0,0 +1,153 @@
+	 */
+	public Collection<AuditLogEntry> getEntries() {
+		return entries.values();
+	}
+
+	/**
+	 * Gets the entries of the log which have a target.
+	 *
+	 * @return The entries of the log which have a target.
+	 * @see TargetedEntry
+	 */
+	public Collection<TargetedEntry> getTargetedEntries() {
+		return entries.values().stream()
+				.filter(TargetedEntry.class::isInstance)
+				.map(TargetedEntry.class::cast)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Gets the entries of the log which have a target which is a Discord object.
+	 *
+	 * @return The entries of the log which have a target which is a Discord object.
+	 * @see DiscordObjectEntry
+	 */
+	public Collection<DiscordObjectEntry<?>> getDiscordObjectEntries() {
+		return entries.values().stream()
+				.filter(DiscordObjectEntry.class::isInstance)
+				.map(entry -> (DiscordObjectEntry<?>) entry)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Gets the entries of the log which have a target which is of the given type.
+	 *
+	 * @param clazz The class of the type of target to search for.
+	 * @param <T> The type of target to search for.
+	 * @return The entries of the log which have a target which is of the given type.
+	 */
+	public <T extends IDiscordObject<T>> Collection<DiscordObjectEntry<T>> getDiscordObjectEntries(Class<T> clazz) {
+		return getDiscordObjectEntries().stream()
+				.filter(entry -> clazz.isAssignableFrom(entry.getTarget().getClass()))
+				.map(entry -> (DiscordObjectEntry<T>) entry)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Gets an entry by its unique snowflake ID.
+	 *
+	 * @param id The ID of the desired entry.
+	 * @return The entry with the given ID (or null if one was not found).
+	 */
+	public AuditLogEntry getEntryByID(long id) {
+		return getEntryMap().get(id);
+	}
+
+	/**
+	 * Gets the entries with the given action type.
+	 *
+	 * @param actionType The action type of the desired entries.
+	 * @return The entries with the given action type.
+	 */
+	public List<AuditLogEntry> getEntriesByType(ActionType actionType) {
+		return getEntries().stream()
+				.filter(entry -> entry.getActionType() == actionType)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Gets the entries which have a target which have the given unique snowflake ID.
+	 *
+	 * @param targetID The ID of the target of the desired entries.
+	 * @return The entries which have a target which have the given unique snowflake ID.
+	 */
+	public List<TargetedEntry> getEntriesByTarget(long targetID) {
+		return getEntries().stream()
+				.filter(TargetedEntry.class::isInstance)
+				.map(TargetedEntry.class::cast)
+				.filter(entry -> entry.getTargetID() == targetID)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * Gets the entries with the given responsible user.
+	 *
+	 * @param user The user responsible for the desired entries.
+	 * @return The entries with the given responsible user.
+	 */
+	public List<AuditLogEntry> getEntriesByResponsibleUser(IUser user) {
```

Comment:
I agree with Panda. If you're pedantic and want it shorter, I think even `getEntriesByActingUser` would work, although I personally like Responsible more.